### ISSUE TITLE
Loading states

### ISSUE THREAD

**Issue by alexcroox** [gabrielcaiana/repository-Pattern-Nuxt-3#1]
Thanks for providing this example. I'm keen to get started with a repository pattern in Nuxt 3.

One thing that keeps tripping me up is how to use the `loading` states from $fetch with this pattern. For example if I want to submit a POST from a form and I want to show a loading state on the submit button.

It feels like I need to initialise the request in setup but then execute it when the form is submitted

**Comment by gabrielcaiana** [comment]
> Thanks for providing this example. I'm keen to get started with a repository pattern in Nuxt 3.
> 
> One thing that keeps tripping me up is how to use the `loading` states from $fetch with this pattern. For example if I want to submit a POST from a form and I want to show a loading state on the submit button.
> 
> It feels like I need to initialise the request in setup but then execute it when the form is submitted

Glad to help with this example

I've been through similar scenarios regarding a global loading, my suggestion in this case is to use a global composable or even a store with pinia, this way whenever a request is triggered you can set this isLoading to true

**Comment by alexcroox** [comment]
Yeah I was hoping to import something when I import/use the repository and have the loading state local to that component rather than having possible 30 global loading states for each endpoint when the rest of the application doesn’t need to know about them. 

I’ll keep plugging away thanks.

**Comment by gabrielcaiana** [comment]
> Yeah I was hoping to import something when I import/use the repository and have the loading state local to that component rather than having possible 30 global loading states for each endpoint when the rest of the application doesn’t need to know about them.
> 
> I’ll keep plugging away thanks.

an example for global use

```
import { defineNuxtPlugin } from '#app'

export default defineNuxtPlugin({
  setup() {
    const api = $fetch.create({
      baseURL: useRuntimeConfig().public.apiUrl,
      onRequest: () => {
       // dispatch global loading
      },
      onResponse: () => {
        // stop global loading
      },
      onRequestError: () => {
       // dispatch global loading
      },
    })

    return { 
      provide: { api }
    }
  }
})
```

I believe that for local loading useAsyncData itself uses a pending, it could be a good solution

https://nuxt.com/docs/api/composables/use-async-data
